google-cloud-clients/google-cloud-bigtable-admin/src/main/java/com/google/cloud/bigtable/admin/v2/BigtableInstanceAdminClient.java

```diff
@@ -417,6 +418,68 @@ public Void apply(Empty input) {
     );
   }
 
+  /**
+   * Checks if the instance specified by the instanceId exists
+   *
+   * <p>Sample code:
+   *
+   * <pre>{@code
+   * if(client.exists("my-instance")) {
+   *   System.out.println("Instance exists");
+   * }
+   * }</pre>
+   */
+  public boolean exists(String instanceId) {
+    return ApiExceptions.callAndTranslateApiException(existsAsync(instanceId));
+  }
+
+  /**
+   * Asynchronously checks if the instance specified by the instanceId exists
+   *
+   * <p>Sample code:
+   *
+   * <pre>{@code
+   * ApiFuture<Boolean> found = client.existsAsync("my-instance");
+   *
+   * ApiFutures.addCallback(
+   *  found,
+   *  new ApiFutureCallback<Boolean>() {
+   *    public void onSuccess(Boolean found) {
+   *      if (found) {
+   *        System.out.println("Instance exists");
+   *      } else {
+   *        System.out.println("Instance not found");
+   *      }
+   *    }
+   *
+   *    public void onFailure(Throwable t) {
+   *      t.printStackTrace();
+   *    }
+   *  },
+   *  MoreExecutors.directExecutor()
+   * );
+   * }</pre>
+   */
+  public ApiFuture<Boolean> existsAsync(String instanceId) {
+    ApiFuture<Instance> protoFuture = getInstanceAsync(instanceId);
+
+    ApiFuture<Boolean> existsFuture = ApiFutures
+        .transform(protoFuture, new ApiFunction<Instance, Boolean>() {
```

Comment:
This ApiFunction could be made into a constant so we don't need to instantiate a new one every call.

Reply:
I'm pretty sure the jvm will optimize away the allocation and I think moving this to a non-local const will decrease readability

Reply:
SGTM